fe/fe-core/src/main/java/com/starrocks/sql/analyzer/BackupRestoreAnalyzer.java

```diff
@@ -84,16 +89,40 @@ public Void visitBackupStatement(BackupStmt backupStmt, ConnectContext context)
                     tableRefs.add(tableRef);
                 }
             }
+
+            Map<Long, TableRef> tableIdToTableRefMap = Maps.newHashMap();
+            Map<String, MaterializedView> mvNameMVMap = Maps.newHashMap();
             for (TableRef tableRef : tableRefs) {
-                analyzeTableRef(tableRef, dbName, database, tblPartsMap, context.getCurrentCatalog());
+                analyzeTableRef(tableRef, dbName, database, tblPartsMap, context.getCurrentCatalog(),
+                        mvNameMVMap, tableIdToTableRefMap);
                 if (tableRef.hasExplicitAlias()) {
                     throw new SemanticException("Can not set alias for table in Backup Stmt: " + tableRef,
                             tableRef.getPos());
                 }
             }
 
-            tableRefs.clear();
-            tableRefs.addAll(tblPartsMap.values());
+            if (!mvNameMVMap.isEmpty()) {
+                // check base table existed in the same db.
+                for (Map.Entry<String, MaterializedView> e : mvNameMVMap.entrySet()) {
+                    MaterializedView mv = e.getValue();
+                    for (BaseTableInfo baseTableInfo : mv.getBaseTableInfos()) {
+                        if (!tableIdToTableRefMap.containsKey(baseTableInfo.getTableId())) {
+                            LOG.warn(String.format("Backup/restore materialized view %s's" +
+                                    " base table %s does not in the same db", mv.getName(), baseTableInfo.getTableId()));
+                        }
+                    }
+                }
+
+                // reorder the tableRefs to ensure the ref table/materialized views are before in the materialized view
```

Comment:
```suggestion
                // reorder the tableRefs to ensure the ref table/materialized views are ahead of the materialized view
```

Reply:
is it a hard limit?

Reply:
yep.  Must reorder it, otherwise we cannot active the materialized view correctly?